radix-engine/src/transaction/state_update_summary.rs

```diff
@@ -386,7 +386,7 @@ impl<'a, S: SubstateDatabase> BalanceAccounter<'a, S> {
         self.tracked
             .get(node_id)
             .and_then(|tracked_node| tracked_node.tracked_modules.get(&module_id))
-            .and_then(|tracked_module| tracked_module.substates.get(&M::to_db_sort_key(key)))
+            .and_then(|tracked_module| tracked_module.substates.get(&key.to_db_sort_key()))
```

Comment:
in this line (in particular) it becomes very visible that introduction of `SortKey` trait defeats the decoupling provided by `mapper` (and also defeats the purpose of mapper trait's existence).

I am open to considering such move, but:
- we should go all the way with it (i.e. do the same to partition key mapping logic!)
- it really is a scope of some next PR, not this one.